src/managers/GitObjectManager.js

```diff
@@ -77,7 +66,24 @@ export class GitObjectManager {
     let { type, object } = GitObject.unwrap({ oid, buffer })
     if (format === 'content') return { type, format: 'content', object, source }
   }
-
+  static async loadPack (fs, filename, getExternalRefDelta) {
+    // If not there, load it from a .idx file
+    const idxName = filename.replace(/pack$/, 'idx')
+    const packFile = fs.read(filename)
```

Comment:
This is a performance bug, because it will cause every single packfile in `.git/objects/pack` to be loaded into memory. We only want to read the packfile into memory if a) we need to create the .idx file from the pack or b) we need to read an object from the packfile.

Reply:
It would be nice to lazy-load it on p.read, but I'm trying to avoid using 'fs' in `models`.